Chore: update main from develop

JustSamuel opened the pull request:
# Description
<!--
What do you want to achieve with this PR? Why did you write this code? What problem does this PR solve?
Describe your changes in detail and, if relevant, explain which choices you have made and why.
When making changes to the UI, make sure to include comparison screenshots!
-->

## Related issues/external references
Style reworks, updated README and other fixes.


## Types of changes
<!-- What types of changes does your code introduce? Remove all the items that do not apply: -->
- Bug fix _(non-breaking change which fixes an issue)_
- New feature _(non-breaking change which adds functionality)_

JustSamuel committed:
feat(pos): style rework (#367)

JustSamuel committed:
feat(pos): style rework (#368)

JustSamuel committed:
docs: update readme (#370)

JustSamuel committed:
ci: add dependabot

dependabot[bot] committed:
chore(deps): update @stripe/stripe-js requirement in /apps/dashboard (#380)

Updates the requirements on [@stripe/stripe-js](https://github.com/stripe/stripe-js) to permit the latest version.
- [Release notes](https://github.com/stripe/stripe-js/releases)
- [Commits](https://github.com/stripe/stripe-js/compare/v4.4.0...v4.8.0)

---
updated-dependencies:
- dependency-name: "@stripe/stripe-js"
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): update @tsconfig/node18 requirement in /apps/dashboard (#383)

---
updated-dependencies:
- dependency-name: "@tsconfig/node18"
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): update axios requirement in /lib/common (#391)

Updates the requirements on [axios](https://github.com/axios/axios) to permit the latest version.
- [Release notes](https://github.com/axios/axios/releases)
- [Changelog](https://github.com/axios/axios/blob/v1.x/CHANGELOG.md)
- [Commits](https://github.com/axios/axios/compare/v1.6.8...v1.7.7)

---
updated-dependencies:
- dependency-name: axios
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): update sass requirement in /apps/point-of-sale (#388)

Updates the requirements on [sass](https://github.com/sass/dart-sass) to permit the latest version.
- [Release notes](https://github.com/sass/dart-sass/releases)
- [Changelog](https://github.com/sass/dart-sass/blob/main/CHANGELOG.md)
- [Commits](https://github.com/sass/dart-sass/compare/1.69.7...1.80.2)

---
updated-dependencies:
- dependency-name: sass
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): update @tsconfig/node18 requirement in /apps/point-of-sale (#387)

---
updated-dependencies:
- dependency-name: "@tsconfig/node18"
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): update axios requirement in /apps/point-of-sale (#386)

Updates the requirements on [axios](https://github.com/axios/axios) to permit the latest version.
- [Release notes](https://github.com/axios/axios/releases)
- [Changelog](https://github.com/axios/axios/blob/v1.x/CHANGELOG.md)
- [Commits](https://github.com/axios/axios/compare/v1.6.8...v1.7.7)

---
updated-dependencies:
- dependency-name: axios
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

JustSamuel committed:
chore: fix pinia typing for pinia 2.2.4 (#392)

dependabot[bot] committed:
chore(deps): update pinia requirement in /lib/common (#390)

Updates the requirements on [pinia](https://github.com/vuejs/pinia) to permit the latest version.
- [Release notes](https://github.com/vuejs/pinia/releases)
- [Commits](https://github.com/vuejs/pinia/compare/pinia@2.1.7...pinia@2.2.4)

---
updated-dependencies:
- dependency-name: pinia
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Samuel <samuel@gewis.nl>

dependabot[bot] committed:
chore(deps): update pinia requirement in /apps/point-of-sale (#385)

Updates the requirements on [pinia](https://github.com/vuejs/pinia) to permit the latest version.
- [Release notes](https://github.com/vuejs/pinia/releases)
- [Commits](https://github.com/vuejs/pinia/compare/pinia@2.1.3...pinia@2.2.4)

---
updated-dependencies:
- dependency-name: pinia
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): update sass requirement in /apps/dashboard (#381)

Updates the requirements on [sass](https://github.com/sass/dart-sass) to permit the latest version.
- [Release notes](https://github.com/sass/dart-sass/releases)
- [Changelog](https://github.com/sass/dart-sass/blob/main/CHANGELOG.md)
- [Commits](https://github.com/sass/dart-sass/compare/1.63.6...1.80.2)

---
updated-dependencies:
- dependency-name: sass
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): update axios requirement in /apps/dashboard (#382)

Updates the requirements on [axios](https://github.com/axios/axios) to permit the latest version.
- [Release notes](https://github.com/axios/axios/releases)
- [Changelog](https://github.com/axios/axios/blob/v1.x/CHANGELOG.md)
- [Commits](https://github.com/axios/axios/compare/v1.6.8...v1.7.7)

---
updated-dependencies:
- dependency-name: axios
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): update eslint-plugin-vue requirement in /apps/dashboard (#394)

Updates the requirements on [eslint-plugin-vue](https://github.com/vuejs/eslint-plugin-vue) to permit the latest version.
- [Release notes](https://github.com/vuejs/eslint-plugin-vue/releases)
- [Commits](https://github.com/vuejs/eslint-plugin-vue/compare/v9.11.0...v9.29.0)

---
updated-dependencies:
- dependency-name: eslint-plugin-vue
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): update yup requirement in /apps/dashboard (#395)

Updates the requirements on [yup](https://github.com/jquense/yup) to permit the latest version.
- [Release notes](https://github.com/jquense/yup/releases)
- [Changelog](https://github.com/jquense/yup/blob/master/CHANGELOG.md)
- [Commits](https://github.com/jquense/yup/compare/v1.2.0...v1.4.0)

---
updated-dependencies:
- dependency-name: yup
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

JustSamuel committed:
chore(deps): bump most deps (#407)

JustSamuel committed:
chore(ci): update dependabot config

JustSamuel committed:
fix: bug in common rootapi (#393)

RubenLWF committed:
feat: Add OpenGraph tags (#389)

SuperVK committed:
refactor(deps): Move dependencies to the top level (#428)

* refactor(dependencies): Move dependencies into root package.json

* refactor(deps): Fix typescript version, remove uuid and add lodash

* refactor(deps): Update yarn.lock

* refactor(dependencies): Post rebase mistakes

* refactor(dependencies): Re-introduce strict mode

* refactor(dependencies): Revert change

JustSamuel committed:
refactor(dashboard): removes deprecated sass functions (#431)

RubenLWF committed:
feat(dashboard): Add maintenance screen (#422)

tomudding committed:
fix(dashboard/user-topup-modal): prevent clicking payment button multiple times (#436)

Pressing the "PAY" button starts the confirmation of the Payment Intent of the
current top-up. This sends a request to Stripe which will (if successful) return
a payment redirect link for iDEAL transactions.

Unfortunately, due to various reasons it is possible to accidentally press the
"PAY" button twice. This can be due to user error, device issues, or network
issues. The result is that sometimes, the request to confirm the Payment Intent
is send twice. This means that two payment redirect links are generated by
Stripe, but only the first one will be used.

Because Stripe has generated the second link (or even more) only that link is
still active for Stripe. In other words, successful payments through the first
link are not actively confirmed by Stripe. This means that that money is
"temporarily" considered lost and the top-up is not successful.

The money will be returned to the user by Stripe within 5 business days,
however, we should do everything in our power to ensure that this does not
happen (especially because people sometimes need to do large top-ups).

This change ensures that the "PAY" button is disabled after pressing it until
the user is redirected to the iDEAL payment page. I would have loved to use an
`Idempotency-Key` instead, however, it appears that this is not directly
supported by the `confirmPayment` method.

SuperVK committed:
feat(fine): Ability to partially waive fines (#429)

* feat(fines): Partially waive fines

* fix(fines): Make mutation refresh after partially waiving fine

* fix(fines): Form validation for waiving above the possible amount of fines

* fix(fines): Fix issues from Samuel

* fix(fines): Add info text for waiving fines

* fix(fines): Add button to waive all fines.

* fix(fines): Typo

* fix(fines): Let the last fine waive be the placeholder

* fix(fines): Don't render waive fines when no fine

* fix(fines): Typing issues

SuperVK commented:
Fine by me to merge, but this will mean that the POS styling is now actuality (which is fine by me)

JustSamuel commented:
> Fine by me to merge, but this will mean that the POS styling is now actuality (which is fine by me)

Yeah I'll change the background later, since that seems to be the "issue"

noelvandevrie committed:
Feature/nfc update (#437)

* feat(dashboard/profile) Initial addition of NFC form

* fix: added locales

added additional locales where they were still missing and improved naming of them.

* feat(dashboard/profile)

Changed where the scanner is opened and removed the dialog and form scanner should now work on mobile chrome browsers

* fix(dashboard/nfc-reader): Addresses remarks in pull request

Addresses remarks made in the pull request by @justSamuel

* fix(dashboard/profile):

removed console.log and whitespace

JustSamuel committed:
fix: remove topup warning on landing page (#438)

JustSamuel committed:
feat: post refactoring invoice ui (#335)
